gh-100474: Fix handling of dirs named index.html in http.server (#100475)
python/cpython: Misc/NEWS.d/next/Library/2022-12-23-21-02-43.gh-issue-100474.gppA4U.rst

> Check that index page is actually a regular file before trying to serve it.
> This avoids issues around directories named "index.html".

Reviewer: ```suggestion
:mod:`http.server` now checks that an index page is actually a regular file before trying
to serve it.  This avoids issues with directories named `index.html`.
```

Reviewer: Changing this because the news entry had no context about where the bug was.

Author: A sensible change.

Reviewer: Thanks for the markup fix — been writing too much markdown!  🙁
